app/controllers/dashboard_controller.rb

```diff
@@ -353,6 +361,13 @@ def widget_zoom
     end
   end
 
+  def widget_refresh
+    assert_privileges("widget_generate_content")
+    w = MiqWidget.find(params[:widget])
+    w.queue_generate_content
+    render :json => {:task => MiqTask.last.id.to_s}, :status => 200
```

Comment:
is it safe to use `MiqTask.last.id.to_s` here, maybe some other process also added a task simultaneously, maybe you could have used the task id that's being returned from `w.queue_generate_content` what do you think?

Reply:
@h-kataria It was temporary until ManageIQ/manageiq#19445 was merged. Now it's using `MiqTask` id created in `queue_generate_content`